programs/bubblegum/src/lib.rs

```diff
@@ -0,0 +1,444 @@
+use anchor_lang::prelude::*;
+use gummyroll::{program::Gummyroll, state::node::Node};
+
+pub mod state;
+pub mod utils;
+
+use crate::state::{
+    leaf_schema::{LeafSchema, RawLeafSchema},
+    metaplex_adapter::MetadataArgs,
+    metaplex_anchor::{MasterEdition, TokenMetadata},
+    Nonce, Voucher,
+};
+use crate::utils::{append_leaf, insert_or_append_leaf, replace_leaf};
+
+declare_id!("BGUMzZr2wWfD2yzrXFEWTK2HbdYhqQCP2EZoPEkZBD6o");
+
+#[derive(Accounts)]
+pub struct InitNonce<'info> {
+    #[account(
+        init,
+        seeds = [b"bubblegum"],
+        payer = payer,
+        space = 8 + 16,
+        bump,
+    )]
+    pub nonce: Account<'info, Nonce>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct CreateTree<'info> {
+    pub tree_creator: Signer<'info>,
+    #[account(
+        seeds = [merkle_roll.key().as_ref()],
+        bump,
+    )]
+    /// CHECK: This account is neither written to nor read from.
+    pub authority: UncheckedAccount<'info>,
+    pub gummyroll_program: Program<'info, Gummyroll>,
+    #[account(zero)]
+    /// CHECK: This account must be all zeros
+    pub merkle_roll: UncheckedAccount<'info>,
```

Comment:
Nit: for consistency across ecosystem, might be worth renaming this to `merkle_slab`, or perhaps more descriptively `merkle_bytes`